core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -2441,9 +2447,8 @@ private SqlNode registerFrom(
               null,
               forceRightNullable,
               lateral);
-      if (newRight != right) {
-        join.setRight(newRight);
-      }
+      // for check style.
+      swapJoinRightNode(join, right, newRight);
```

Comment:
Don't think this is a right change, you should make the `newRight` correct first.

Reply:
good catch